modules/mysql/src/test/java/org/testcontainers/containers/MySQLRootAccountTest.java

```diff
@@ -0,0 +1,52 @@
+package org.testcontainers.containers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+@Slf4j
+@RunWith(Parameterized.class)
+public class MySQLRootAccountTest {
+
+    @Parameterized.Parameters(name = "{0}")
+    public static String[] params() {
+        return new String[]{
+            "mysql:8",
+            "mysql:5"
+        };
+    }
+
+    @Parameterized.Parameter()
+    public String image;
+
+    @Test
+    public void testRootAccountUsageWithDefaultPassword() throws SQLException {
+        testWithDB(new MySQLContainer<>(image).withUsername("root"));
+    }
+
+    @Test
+    public void testRootAccountUsageWithEmptyPassword() throws SQLException {
+        testWithDB(new MySQLContainer<>(image).withUsername("root").withPassword(""));
+    }
+
+    @Test
+    public void testRootAccountUsageWithCustomPassword() throws SQLException {
+        testWithDB(new MySQLContainer<>(image).withUsername("root").withPassword("not-default"));
+    }
+
+    private void testWithDB(MySQLContainer<?> db) throws SQLException {
+        try {
+            db.withLogConsumer(new Slf4jLogConsumer(log)).start();
+            Connection connection = DriverManager.getConnection(db.getJdbcUrl(), db.getUsername(), db.getPassword());
+            connection.createStatement().execute("SELECT 1");
```

Comment:
I'd like to add this so that we can be sure we're actually using the root account (e.g. we might think we're using root, but we might be getting a non-root account):
```suggestion
            connection.createStatement().execute("SELECT 1");
            connection.createStatement().execute("set sql_log_bin=0"); // requires root
```